src/main/java/com/minecolonies/coremod/colony/requestsystem/requests/StandardRequestFactories.java

```diff
@@ -0,0 +1,283 @@
+package com.minecolonies.coremod.colony.requestsystem.requests;
+
+import com.google.common.reflect.TypeToken;
+import com.minecolonies.api.colony.requestsystem.RequestState;
+import com.minecolonies.api.colony.requestsystem.factory.IFactoryController;
+import com.minecolonies.api.colony.requestsystem.request.IRequestFactory;
+import com.minecolonies.api.colony.requestsystem.requestable.Delivery;
+import com.minecolonies.api.colony.requestsystem.requester.IRequester;
+import com.minecolonies.api.colony.requestsystem.token.IToken;
+import com.minecolonies.api.util.constant.Suppression;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.nbt.NBTTagInt;
+import net.minecraft.nbt.NBTTagList;
+import net.minecraftforge.common.util.Constants;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Final class holding all the factories for requestables inside minecolonie
+ */
+public final class StandardRequestFactories
+{
+
+    ////// --------------------------- NBTConstants --------------------------- \\\\\\
+    private static final String NBT_TOKEN     = "Token";
+    private static final String NBT_STATE     = "State";
+    private static final String NBT_REQUESTED = "Requested";
+    private static final String NBT_RESULT    = "Result";
+    private static final String NBT_PARENT    = "Parent";
+    private static final String NBT_CHILDREN  = "Children";
+    ////// --------------------------- NBTConstants --------------------------- \\\\\\
+
+    /**
+     * Private constructor to hide the implicit public one.
+     */
+    private StandardRequestFactories()
+    {
+    }
+
+    @SuppressWarnings(Suppression.BIG_CLASS)
+    public static final class ItemStackFactory implements IRequestFactory<ItemStack, StandardRequests.ItemStackRequest>
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this class from 80 to the maximum allowed 30 or externalize it in a public class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)